### ISSUE TITLE
[FR] Combine RAG with search engine of Obsidian

### ISSUE THREAD

**Issue by wwjCMP** [glowingjade/obsidian-smart-composer#129]
Combine RAG with search engine of Obsidian in Valut chat mode because RAG usually can not work well when database contain a large amount of irrelevant content.

**Comment by kevin-on** [comment]
Could you elaborate on the implementation details for integrating the Obsidian search engine with RAG? Specifically, are you proposing to run RAG only on the subset of documents returned by Obsidian's search results, rather than the entire database?

**Comment by wwjCMP** [comment]
> Could you elaborate on the implementation details for integrating the Obsidian search engine with RAG? Specifically, are you proposing to run RAG only on the subset of documents returned by Obsidian's search results, rather than the entire database?

Yes, that's the idea. Of course, this should be optional. If the user has a clear understanding of the scope of the answer to the question, this method should be used to allow the user to specify the RAG retrieval scope in advance. If the user is not clear about the exact scope of the answer to the question, it should be handled according to the general RAG method.

**Comment by glowingjade** [comment]
If we use the Obsidian search engine, what kind of query do you think should go into it?

**Comment by wwjCMP** [comment]
> If we use the Obsidian search engine, what kind of query do you think should go into it?

Considering the two common ways most people use Obsidian for knowledge management, tags and paths are very likely, and of course, keywords will also be used. It all depends on each individual's usage habits and specific question.

**Comment by glowingjade** [comment]
Actually, we are planning to implement tag mentions! (e.g. write `@tag:abc` and it will automatically retrieve documents with that tag)
Keyword mentions could also be valuable - we'll consider implementing that alongside tag mentions. Thanks for the great suggestion!
